Source/Transforms/LMCComplexFFT.cc

```diff
@@ -248,13 +172,52 @@ namespace locust
         	LERROR(lmclog,"Couldn't make FIR filter causal");
         	exit(-1);
         }
+        
         for (int i = 0; i < fSize+2*fNShiftBins; ++i)
         {
             out[i][0]=fOutputArray[i][0]*2/fTotalWindowSize;
         }
+        
+        return true;
+    }
+
+    bool ComplexFFT::ApplyWindowFunction(int size, fftw_complex* in)
+    {
+        if(!fWindowFunction.IsWindowGenerated())
+        {
+            LERROR(lmclog,"Error applying window function, it has not been generated");
+            exit(-1); 
+        }
+        double windowfactor = 0.0;
+        for (int i = 0; i < size; ++i)
+        { 
+            windowfactor = fWindowFunction.GetWindowFunction()->at(i+fPreFilterBins);
```

Comment:
I think you may have to do this here:
```const aWindowFunction=fWindowFunction.GetWindowFunction()->at(i+fPreFilterBins);```
outside the function and then inside the function 
```windowfactor = aWindowFunction.GetWindowFunction()->at(i+fPreFilterBins);```
This will also reduce the calls to `GetWindowFunction` for every `i` anyway.